app/routes.py

```diff
@@ -1,2 +1,274 @@
+    if not task:
+        return make_response(f"Task #{task_id} Not Found", 404)
+
+    task.completed_at = None
+    db.session.commit()
+
+    task_response = {"task": {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": False
+    }}
+    return make_response(task_response, 200)
+
+
+@goals_bp.route("/", methods=["GET", "POST"])
+def handle_goals():
+    if request.method == "GET":
+        goals = Goal.query.all()
+
+        goals_response = []
+
+        for goal in goals:
+            goals_response.append({
+                "id": goal.goal_id,
+                "title": goal.title
+            })
+        return make_response(jsonify(goals_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if 'title' not in request_body:
+            goal_response = {"details": "Invalid data"}
+            return make_response(goal_response, 400)
+
+        new_goal = Goal(title=request_body["title"])
+
+        db.session.add(new_goal)
+        db.session.commit()
+
+        goal_response = {"goal":
+                         {"id": new_goal.goal_id,
+                          "title": new_goal.title,
+                          }}
+        return make_response(goal_response, 201)
+
+
+@goals_bp.route("/<goal_id>", methods=["GET", "PUT", "DELETE"])
+def handle_goal(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return make_response(f"Goal #{goal_id} Not Found", 404)
+
+    if request.method == "GET":
+        goal_response = {"goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}
+        return make_response(goal_response, 200)
+
+    elif request.method == "PUT":
+        form_data = request.get_json()
+        if not 'title' in form_data:
+            goal_response = {"details": "Invalid data"}
+            return make_response(goal_response, 400)
+
+        goal.title = form_data["title"]
+        db.session.commit()
+
+        goal_response = {"goal": {
+            "id": goal.goal_id,
+            "title": goal.title
+        }}
+        return make_response(goal_response, 200)
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+        goal_response = {
+            "details": f"Goal {goal.goal_id} \"{goal.title}\" successfully deleted"
+        }
+        return make_response(goal_response, 200)
+
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST", "GET"])
+def handle_goal_tasks(goal_id):
+    goal = Goal.query.get(goal_id)
+    if not goal:
+        return make_response(f"Goal #{goal_id} Not Found", 404)
+
+    if request.method == "POST":
+        request_body = request.get_json()
+        task_ids = request_body["task_ids"]
+        for task_id in task_ids:
+            task = Task.query.get(task_id)
+            # if not task:
+            #     return make_response(f"Task #{task_id} Not Found", 404)
```

Comment:
Curious why you commented this out. We _should_ check to make sure that each task id we lookup is valid.